### ISSUE TITLE
'Disk full' condition wiped out the chat history

### ISSUE THREAD

**Issue by yurivict** [qTox/qTox#2179]
Reboot in 'disk full' condition left the chat history blank.

I don't understand how this is possible, because supposedly SqLite is used, and it isn't supposed to lose data like this. Unless you somehow rewrite the whole file during the save operation.

I am actually not sure why tox people asked me to create bug here. It should be in toxcore. Should be an API into toxcore like: 1. getUserHistory(...) 2. addHistoryMessage(...) It should all reside in toxcore IMO

**Comment by zetok** [comment]
@yurivict could you provide steps to reproduce?

**Comment by yurivict** [comment]
I believe SQLite has a bug that causes the data loss on disk full condition. I can't provide specific steps beyond staging the disk-full condition and trying to use qTox.

**Comment by zetok** [comment]
>I can't provide specific steps beyond staging the disk-full condition and trying to use qTox.

I couldn't reproduce history being erased under those conditions.

**Comment by yurivict** [comment]
I had the data loss in the unrelated app using SQLite too. I am pretty sure this is an SQLite problem.

**Comment by zetok** [comment]
Steps?

**Comment by yurivict** [comment]
Ok, I will close this because this isn't a qTox bug.

**Comment by zetok** [comment]
>Ok, I will close this because this isn't a qTox bug.

There's no problem with keeping issues for qTox dependencies open as long as they're affecting qTox and haven't been fixed in the dependency.

The problem here is that the issue sounds like something that should be reproducible, but I couldn't reproduce it when I've tried.

I've found some other problems (#3946), but ho history loss. Now, there was message loss, but only of message that wasn't saved due to no free space – messages that were saved before space ran out weren't affected.
